cardano-cli/src/Cardano/CLI/EraBased/Query/Command.hs

```diff
@@ -253,6 +255,13 @@ data QueryTreasuryValueCmdArgs era = QueryTreasuryValueCmdArgs
   }
   deriving Show
 
+data QueryEraHistoryCmdArgs era = QueryEraHistoryCmdArgs
+  { eon :: !(ConwayEraOnwards era)
```

Comment:
Why is this restricted to Conway?

Reply:
But maybe it is already not necessary to worry about it and just use `EraParamsWithGenesisWindow`

Reply:
It doesn't need to be. This query can be in any shelley based era.

Reply:
Addressed here: https://github.com/IntersectMBO/cardano-cli/pull/1079/commits/80dbabb4e7ceebccbf89c5d408185b0d5bda3007